x-pack/plugins/triggers_actions_ui/public/application/components/builtin_action_types/es_index/es_index_params.tsx

```diff
@@ -20,38 +34,152 @@ export const IndexParamsFields = ({
+    ALERT_HISTORY_PREFIX,
+    ''
+  );
+
+  const getDocumentToIndex = (doc: Array<Record<string, any>> | undefined) =>
+    doc && doc.length > 0 ? ((doc[0] as unknown) as string) : undefined;
+
+  const [documentToIndex, setDocumentToIndex] = useState<string | undefined>(
+    getDocumentToIndex(documents)
+  );
+  const [alertHistoryIndexSuffix, setAlertHistoryIndexSuffix] = useState<string>(
+    indexOverride ? indexOverride.replace(ALERT_HISTORY_PREFIX, '') : defaultAlertHistoryIndexSuffix
+  );
+  const [usePreconfiguredSchema, setUsePreconfiguredSchema] = useState<boolean>(false);
+
+  useEffect(() => {
+    setDocumentToIndex(getDocumentToIndex(documents));
+  }, [documents]);
+
+  useEffect(() => {
+    if (actionConnector?.id === AlertHistoryEsIndexConnectorId) {
+      setUsePreconfiguredSchema(true);
+      editAction('documents', [JSON.stringify(AlertHistoryDocumentTemplate)], index);
+      setDocumentToIndex(JSON.stringify(AlertHistoryDocumentTemplate));
+    } else {
+      setUsePreconfiguredSchema(false);
+      editAction('documents', undefined, index);
+      setDocumentToIndex(undefined);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [actionConnector?.id]);
 
   const onDocumentsChange = (updatedDocuments: string) => {
     try {
       const documentsJSON = JSON.parse(updatedDocuments);
       editAction('documents', [documentsJSON], index);
+      setDocumentToIndex(updatedDocuments);
     } catch (e) {
       // set document as empty to turn on the validation for non empty valid JSON object
       editAction('documents', [{}], index);
+      setDocumentToIndex(undefined);
     }
   };
 
-  return (
+  const documentsFieldLabel = i18n.translate(
+    'xpack.triggersActionsUI.components.builtinActionTypes.indexAction.documentsFieldLabel',
+    {
+      defaultMessage: 'Document to index',
+    }
+  );
+
+  const resetDefaultIndex =
+    indexOverride && indexOverride !== AlertHistoryDefaultIndexName ? (
+      <EuiText size="xs">
+        <EuiLink
+          data-test-subj="resetDefaultIndex"
+          onClick={() => {
+            editAction('indexOverride', AlertHistoryDefaultIndexName, index);
+            setAlertHistoryIndexSuffix(defaultAlertHistoryIndexSuffix);
+          }}
+        >
+          <EuiIcon type="refresh" />
+          <FormattedMessage
+            id="xpack.triggersActionsUI.sections.alertsList.resetDefaultIndexLabel"
+            defaultMessage="Reset default index"
+          />
+        </EuiLink>
+      </EuiText>
+    ) : (
+      <></>
+    );
+
+  const preconfiguredDocumentSchema = (
+    <>
+      <EuiFormRow
+        fullWidth
+        error={errors.indexOverride as string[]}
+        isInvalid={(errors.indexOverride as string[]) && errors.indexOverride.length > 0}
+        label={i18n.translate(
+          'xpack.triggersActionsUI.components.builtinActionTypes.indexAction.preconfiguredIndex',
+          {
+            defaultMessage: 'ES index',
```

Comment:
Similar question regarding: https://github.com/elastic/kibana/issues/93406
```suggestion
            defaultMessage: 'Elasticsearch index',
```

Reply:
Updated in https://github.com/elastic/kibana/pull/94909/commits/5aedddb16b8ab2fe9d1f090a1aef7eea0c7c36d0